server/extension/extension.go

```diff
@@ -580,17 +646,19 @@ func (m *Manager) CallExtension(extName string, registry ProxyRegistry) func(htt
 			return
 		}
 
-		sanitizeRequest(r, extName)
+		prepareRequest(r, extName, app)
 		m.log.Debugf("proxing request for extension %q", extName)
 		proxy.ServeHTTP(w, r)
 	}
 }
 
-// sanitizeRequest is reponsible for preparing and cleaning the given
+// prepareRequest is reponsible for preparing and cleaning the given
 // request, removing sensitive information before forwarding it to the
 // proxy extension.
-func sanitizeRequest(r *http.Request, extName string) {
+func prepareRequest(r *http.Request, extName string, app *v1alpha1.Application) {
 	r.URL.Path = strings.TrimPrefix(r.URL.Path, fmt.Sprintf("%s/%s", URLPrefix, extName))
-	r.Header.Del("Cookie")
-	r.Header.Del("Authorization")
+	r.Header.Set(HeaderArgoCDTargetClusterURL, app.Spec.Destination.Server)
+	if app.Spec.Destination.Name != "" {
+		r.Header.Set(HeaderArgoCDTargetClusterName, app.Spec.Destination.Name)
+	}
```

Comment:
```suggestion
	if app.Spec.Destination.Server != "" {
		r.Header.Set(HeaderArgoCDTargetClusterURL, app.Spec.Destination.Server)
	}
	if app.Spec.Destination.Name != "" {
		r.Header.Set(HeaderArgoCDTargetClusterName, app.Spec.Destination.Name)
	}
```

Maybe? Just so they behave the same.

Reply:
makes sense. Fixed